v2/tools/generator/internal/astmodel/kubebuilder_validations.go

```diff
@@ -158,11 +157,7 @@ func MakeMaximumValidation(value *big.Rat) KubeBuilderValidation {
 	if value.IsInt() {
 		return KubeBuilderValidation{MaximumValidationName, value.RatString()}
 	} else {
-		floatValue, ok := value.Float64()
```

Comment:
I don't see it happening either, which (I think) makes it safe to remove.

Reply:
My concern here is that we're dropping an `ok` on the floor here, and while it may be true that right now it doesn't occur, there's nothing stopping it from happening in the future. If it did... do we care?

I'm not totally clear but I think we should either:
1. Panic on `!ok`
2. Have a detailed comment explaining why it's ok to ignore the `ok` result.

I tend to err on the side of 1. I'd rather panic here than have some wrong min/max and not know about it.

Reply:
Same applies below too.

Reply:
*I'd be fine with returning an error here too but that might make various callers have to do a lot of work to deal with it.

Reply:
Restoring the check makes sense, but there's a lot of infrastructure required first. I've added a note on this to #2969 and included this comment in all three locations:

```
//TODO (@bearps) Restore this check when we modify AsDeclarations() to allow error returns
```

Reply:
Based on offline discussion, have also put `panic()` calls in all three locations.